DependencyInjection/HttplugExtension.php

```diff
@@ -162,4 +168,53 @@ private function configurePluginByName($name, Definition $definition, array $con
                 break;
         }
     }
+
+    /**
+     * @param ContainerBuilder $container
+     * @param Definition       $parent
+     * @param array            $config
+     */
+    private function configureAuthentication(ContainerBuilder $container, array $config)
+    {
+        foreach ($config as $name => $values) {
+            $authServiceKey = sprintf('httplug.plugin.authentication.%s.auth', $name);
+            if ($values['type'] === 'bearer') {
+                $this->validateAuthenticationConfiguration($values, 'bearer', $name, ['token']);
+                $container->register($authServiceKey, Bearer::class)
+                    ->addArgument($values['token']);
+            } elseif ($values['type'] === 'basic') {
+                $this->validateAuthenticationConfiguration($values, 'bearer', $name, ['username', 'password']);
+                $container->register($authServiceKey, BasicAuth::class)
+                    ->addArgument($values['username'])
+                    ->addArgument($values['password']);
+            } elseif ($values['type'] === 'wsse') {
+                $this->validateAuthenticationConfiguration($values, 'bearer', $name, ['username', 'password']);
+                $container->register($authServiceKey, Wsse::class)
+                    ->addArgument($values['username'])
+                    ->addArgument($values['password']);
+            } else {
+                throw new \LogicException(sprintf('Unknown authentication type: "%s"', $values['type']));
+            }
+
+            $container->register('httplug.plugin.authentication.'.$name, AuthenticationPlugin::class)
+                ->addArgument(new Reference($authServiceKey));
+        }
+    }
+
+    /**
+     * Throw a Logic error with a descriptive error if some configuration is missing.
+     *
+     * @param array  $config
+     * @param string $authType
+     * @param string $type
+     * @param array  $names
+     */
+    private function validateAuthenticationConfiguration(array $config, $type, $authType, array $names)
```

Comment:
Great! Thank you for this example. I will remove the `validateAuthenticationConfiguration` and use something like this instead.